angr/analyses/decompiler/structured_codegen/c.py

```diff
@@ -2767,9 +2780,27 @@ def _handle_Condition(self, condition_node: ConditionNode, **kwargs):
 
         else_node = self._handle(condition_node.false_node, is_expr=False) if condition_node.false_node else None
 
+        true_node = condition_node.true_node
+        do_simplify_else = False
+
+        if self.simplify_else_scope:
+            last_block = None
```

Comment:
Make this entire block a boolean method in `c.py` (put it near the to). Maybe call it:
```python
def is_simple_return_node(node: Union[ailment.Block, SequenceNode]) -> bool:
    ...
```